.github/workflows/check-api-changes.yaml

```diff
@@ -0,0 +1,101 @@
+name: "Check API Changes"
+
+on:
+  pull_request:
+    branches: [ "main" ]
+    types:
+      - "opened"
+      - "reopened"
+      - "synchronize"
+      - "labeled"
+      - "unlabeled"
+
+jobs:
+  check-api-diff:
+    if: ${{ !contains(github.event.pull_request.labels.*.name, 'confirm/api-changes') }}
+    name: Diff CRDs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout PR ref of lifecycle-manager
+        uses: actions/checkout@v4
+
+      - name: Copy PR CRDs to temp dir
+        run: mkdir -p /tmp/pr-crds && cp ./config/crd/bases/* /tmp/pr-crds/
+
+      - name: Checkout main of lifecycle-manager
+        uses: actions/checkout@v4
+        with:
+          ref: main
+
+      - name: Copy main CRDs to temp dir
+        run: mkdir -p /tmp/main-branch-crds && cp ./config/crd/bases/* /tmp/main-branch-crds/
```

Comment:
that was my first instict but the "main" in main-crds felt like a modifier of the crds instead of referring to the branch. But I guess, that was just overthinking